packages/astro/src/runtime.ts

```diff
@@ -214,7 +214,7 @@ async function load(config: RuntimeConfig, rawPathname: string | undefined): Pro
       request: {
         // params should go here when implemented
         url: requestURL,
-        canonicalURL: canonicalURL(requestURL.pathname, requestURL.origin),
+        canonicalURL: canonicalURL(requestURL.pathname, site.toString()),
```

Comment:
Is site not a URL? I'm pretty sure you can just use a URL as the param rather than a string 🤔

Reply:
This takes a string because the second arg is optional and it's used in some places that don't normalize the user-provided site to a URL.